articles/libraries/auth0-spa-js/index.md

```diff
@@ -169,6 +185,49 @@ document.getElementById('logout').addEventListener('click', () => {
 });
 ```
 
+### Change storage options
+
+The Auth0 SPA SDK stores tokens in memory by default. However, this does not provide persistence across page refreshes and browser tabs. Instead, you can opt-in to store tokens in local storage by setting the `cacheLocation` property to `localstorage` when initializing the SDK.
+
+::: warning
+Storing tokens in browser local storage provides persistence across page refreshes and browser tabs. However, if an attacker can achieve running JavaScript in the SPA using a cross-site scripting (XSS) attack, they can retrieve the tokens stored in local storage. A vulnerability leading to a successful XSS attack can be either in the SPA source code or in any third-party JavaScript code (such as bootstrap, jQuery, or Google Analytics) included in the SPA.
+
+Read more about [token storage](/tokens/concepts/token-storage#single-page-app-scenarios).
+:::
+
+```js
+const auth0 = await createAuth0Client({
+  domain: '${account.namespace}',
+  client_id: '${account.clientId}',
+  cacheLocation: 'localstorage'
+});
+```
+
+### Use rotating Refresh Tokens
+
+The Auth0 SPA SDK can be configured to use [rotating Refresh Tokens](/tokens/concepts/refresh-token-rotation) to get new access tokens silently. These can be used to bypass browser privacy technology that prevents access to the Auth0 session cookie when authenticating silently, as well as providing [built-in reuse detection](/tokens/concepts/refresh-token-rotation#automatic-reuse-detection).
```

Comment:
I wonder if "These can be used to bypass browser privacy technology that prevents access to the Auth0 session cookie when authenticating silently" should be mentioned further up since the `localstorage` feature was also added to mitigate this?

Reply:
Yes can definitely add some words to that effect. Local storage was partly about mitigating those effects, but more about the fact that storing refresh tokens short-term in memory doesn't really make a lot of sense. Depends on the use case though.

I'll add something.